scripts/make-release.sh

```diff
@@ -63,12 +64,22 @@ git tag -a $git_tag -m "Release $git_tag"
 echo Tagged release with $git_tag
 
 # Update version for next build
-# Assumes version is in semver form and last number is preceded by a '.' character
+# Increment last number-only component and add '-next' if not already there.
+# e.g. 1.2.3            -->  1.2.4-next
+#      1.2.3-rc.4       -->  1.2.3-rc.5-next
+#      1.2.3-rc.4-next  -->  1.2.3-rc.5-next
 cd $chart_dir
-prefix=${version%.*}
-last_component=${version##*.}
-((last_component++))
-next_version="$prefix.$last_component"
+next_version=$( echo $version | sed -E -e 's/(.*\.)([0-9]+)(\.([[:alnum:]]*[[:alpha:]-]+[[:alnum:]-]*))?/\1 \2 \3/' \
```

Comment:
Is there a simpler way to do this? If not... it needs a lot of comments to explain it :).

Reply:
Probably, but documented...